gen/field.go

```diff
@@ -53,12 +57,26 @@ func (f fieldGroupGenerator) DefineStruct(g Generator) error {
 		`type <.Name> struct {
 			<range .Fields>
 				<if .Required>
-					<goCase .Name> <typeReference .Type>
+					<goCase .Name> <typeReference .Type> <tag .>
 				<else>
-					<goCase .Name> <typeReferencePtr .Type>
+					<goCase .Name> <typeReferencePtr .Type> <tag .>
 				<end>
 			<end>
-		}`, f)
+		}`,
+		f,
+		TemplateFunc("tag", func(f *compile.FieldSpec) string {
+			// We want to add omitempty if the field is a struct or if it's an
+			// optional primitive so that we don't have "null" noise in the
+			// output. Note that binary is not a primitive.
+
+			if isStructType(f.Type) || (isPrimitiveType(f.Type) && !f.Required) {
+				return fmt.Sprintf("`json:\"%s,omitempty\"`", f.Name)
```

Comment:
required structs should not need `omitempty` either since they should always be set, and if not, i think we want to see "null" so we know it's not set